### PR TITLE
Bug 2067076: show message when there are more than 1 host in sno clusters

### PR DESCRIPTION
**Pull request by ammont82**
Related to https://github.com/openshift-assisted/assisted-ui-lib/pull/1270

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts [hunk 1]
```diff
@@ -161,12 +161,13 @@ export const onHostsNext = async ({ values, clusterDeployment, agents, agentClus
 }
 
 /** AI-specific version for the CIM-flow's onHostsNext() callback */
-export const onDiscoveryHostsNext = async ({ clusterDeployment, agents, agentClusterInstall }: OnDiscoverHostsNext) => {
+export const onDiscoveryHostsNext = async ({ values,clusterDeployment, agents, agentClusterInstall }: OnDiscoverHostsNext) => {
     // TODO(mlibra): So far we do not need "values" of the Formik - options the user will choose from will come later (like CNV or OCS)
     // So far no need to "release" agents since the user either deletes the agent or keep the list untouched. Reconsider when "disable" gets in place.
```
Comment (rawagner) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts:166]: can you remove these comments too ?

Comment (ammont82) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts]: done!

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts [hunk 2]
```diff
@@ -161,12 +161,11 @@ export const onHostsNext = async ({ values, clusterDeployment, agents, agentClus
...
-    // So far no need to "release" agents since the user either deletes the agent or keep the list untouched. Reconsider when "disable" gets in place.
-
+export const onDiscoveryHostsNext = async ({ values,clusterDeployment, agents, agentClusterInstall }: OnDiscoverHostsNext) => {
+    const hostIds =  values.selectedHostIds
     const name = clusterDeployment.metadata.name
     const namespace = clusterDeployment.metadata.namespace
+    agents = agents.filter((a) => !hostIds.includes(a.metadata.uid))
 
     await addAgentsToCluster({
         agents,
```
Comment (rawagner) [inline frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts:171]: Lets not assign back to `agents`. Not a good practice. We can do the filtering inline

```suggestion
    const name = clusterDeployment.metadata.name
    const namespace = clusterDeployment.metadata.namespace

    await addAgentsToCluster({
        agents: agents.filter((a) => !values.selectedHostIds.includes(a.metadata.uid))
```

**Review by rawagner — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts (hunk 2 above)
Comment (ammont82) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/CreateCluster/components/assisted-installer/utils.ts]: Changed!

**Comment by jtomasek** [comment]
Included in https://github.com/stolostron/console/pull/1468

**Pull request closed without merging**